connections.md

```diff
@@ -1,8 +1,252 @@
+
+The message is then transmitted to the `serviceEndpoint`.
+
+We are now in the `requested` state.
+
+##### Request processing
+
+After receiving the exchange request, the _inviter_ evaluates the provided DID and DID Doc according to the DID Method Spec.
+
+The _inviter_ should check the information presented with the keys used in the wire-level message transmission to ensure they match.
+
+The _inviter_ MAY look up the corresponding invitation identified in the request's `~thread.pthid` to determine whether it should accept this exchange request.
+
+If the _inviter_ wishes to continue the exchange, they will persist the received information in their wallet. They will then either update the provisional service information to rotate the key, or provision a new DID entirely. The choice here will depend on the nature of the DID used in the invitation.
+
+The _inviter_ will then craft an exchange response using the newly updated or provisioned information.
+
+##### Request Errors
+
+See [Error Section](#errors) above for message format details.
+
+**request_rejected**
+
+Possible reasons:
+
+- Unsupported DID method for provided DID
+- Expired Invitation
+- DID Doc Invalid
+- Unsupported key type
+- Unsupported endpoint protocol
+- Missing reference to invitation
+
+**request_processing_error**
+
+- unknown processing error
+
+#### 2. Exchange Response
+
+The exchange response message is used to complete the exchange. This message is required in the flow, as it updates the provisional information presented in the invitation.
+
+##### Example
+
+```json
+{
+  "@type": "https://didcomm.org/didexchange/1.0/response",
+  "@id": "12345678900987654321",
+  "~thread": {
+    "thid": "<The Thread ID is the Message ID (@id) of the first message in the thread>"
+  },
+  "did": "B.did@B:A",
+  "did_doc~attach": {
+    "mime-type": "",
+    "data": {
+      "base64": "<b64 encoded bytes of diddoc>"// DID Doc contents here.
+    }
+    "sig": <sig info>
+  }
+}
+```
+
+The signature data must be used to verify against the invitation's `recipientKeys` for continuity.
+
+##### Attributes
+
+* The `@type` attribute is a required string value that denotes that the received message is an exchange request.
+* The `~thread` block contains a `thid` reference to the `@id` of the request message.
+* The `connection` attribute contains the `did` and `did_doc` attributes to enable simpler signing.
+* The `did` attribute is a required string value and denotes DID in use by the _inviter_. Note that this may not be the same DID used in the invitation.
+* The `did_doc~attach` attribute contains the associated DID Doc. If the DID method for the presented DID is not a peer method and the DID Doc is resolvable on a ledger, the `did_doc` attribute is optional.
```

Comment:
is the use of `~attach` significant here? I remember seeing something about this syntax in previous aries RFCs... if this is already in use, I'm very much in favor of not changing it and creating extra work, just want to make sure I know when to suggest the use of `~`.

Reply:
The ~attach stuff is a decorator naming convention. We'll probably be revising decorators in Aries into message headers in DIDComm, so this convention may become less relevant before we're done.